### PR TITLE
Apply code fixes from recently added analyzers

### PR DESCRIPTION
**Pull request by pgovind**
Code-fixes triggered by https://github.com/dotnet/roslyn-analyzers/pull/3610 and https://github.com/dotnet/runtime/issues/33786

I've been careful to make changes only on the projects that target netstandard2.1+ and add the analyzer to the `NoWarn` tags for the ones that also target 2.0.

### CODE REVIEW

File: src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@
 using System.Security.AccessControl;
 using System.Text;
 
+#pragma warning disable CA2249 // Consider using 'string.Contains' instead of 'string.IndexOf'
```
Comment (pgovind) [inline src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs:14]: If we'd rather put this line close to the where the rule is actually triggered, let me know!

File: src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs [hunk 2]
```diff
@@ -1144,7 +1144,7 @@ public StringBuilder Remove(int startIndex, int length)
             return this;
         }
 
-        public StringBuilder Append(bool value) => Append(value.ToString());
+        public StringBuilder Append(bool value) => Append(value);
```
Comment (danmosemsft) [inline src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs:1147]: Recursion

Comment (pgovind) [reply src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs:1147]: Ugh, I thought I removed that. Fixed

File: src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs [hunk 3]
```diff
@@ -106,8 +106,7 @@ internal static void AppendKeyValuePairBuilder(StringBuilder builder, string key
                     // <value> -> <value>
                     builder.Append(keyValue);
                 }
-                // string.Contains(char) is .NetCore2.1+ specific
-                else if ((-1 != keyValue.IndexOf('\"')) && (-1 == keyValue.IndexOf('\'')))
+                else if ((-1 != keyValue.IndexOf('\"')) && (!keyValue.Contains('\'')))
```
Comment (ANahr) [inline src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs:109]: Why replace one but not the other?

File: src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs:14]: Yes, we should surround the relevant code with:
```C#
#pragma warning disable CA2249
...
#pragma warning restore CA2249
```
That said, why does it need to be suppressed?

File: src/libraries/System.Data.Odbc/src/Common/System/Data/Common/AdapterUtil.Odbc.cs [hunk 4]
```diff
@@ -10,6 +10,8 @@
 using System.Runtime.Versioning;
 using System.Text;
 
+#pragma warning disable CA2249 // Consider using 'string.Contains' instead of 'string.IndexOf'
```
Comment (stephentoub) [inline src/libraries/System.Data.Odbc/src/Common/System/Data/Common/AdapterUtil.Odbc.cs:13]: We're suppressing it because it builds for some configurations that lack the relevant overload?  Would it be better then to suppress the rule with NoWarn for the whole project?

File: src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs [hunk 5]
```diff
@@ -95,8 +95,7 @@ public bool UseDefaultCredentials
         }
 
         private static Uri? CreateProxyUri(string? address) =>
-            address == null ? null :
-            address.IndexOf("://", StringComparison.Ordinal) == -1 ? new Uri("http://" + address) :
+            address == null ? null : !address.Contains("://") ? new Uri("http://" + address) :
```
Comment (stephentoub) [inline src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs:98]: Did you change this formatting or did the auto-fixer?  If the latter, looks like we need to fix the fixer's handling of trivia.  Regardless, can you please add back the newline before `!address`?

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs [hunk 6]
```diff
@@ -6,6 +6,8 @@
 using System.Collections.Immutable;
 using System.Reflection.Metadata;
 
+// #pragma warning disable CA2249 // Consider using 'string.Contains' instead of 'string.IndexOf'
```
Comment (stephentoub) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs:9]: Why was this line added?

File: src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs (hunk 5 above)
Comment (pgovind) [reply src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs:98]: The auto-fix did. Good catch on the trivia. I fixed it in the analyzer :)

File: src/libraries/System.Data.Odbc/src/Common/System/Data/Common/AdapterUtil.Odbc.cs (hunk 4 above)
Comment (pgovind) [reply src/libraries/System.Data.Odbc/src/Common/System/Data/Common/AdapterUtil.Odbc.cs:13]: Yes for the first question. I'll add the rule to the nowarn tags in the csproj

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs (hunk 6 above)
Comment (pgovind) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs:9]: I was testing the build :). Removed now

**Comment by pgovind** [comment]
Friendly bump. Can I get a stamp on this please?

File: src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj [hunk 7]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <AssemblyName>System.Net.WebSockets</AssemblyName>
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
-    <NoWarn>$(NoWarn);CS1573</NoWarn>
+    <NoWarn>$(NoWarn);CS1573;CA2249</NoWarn>
```
Comment (stephentoub) [inline src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj:5]: Why was this rule disabled here?

File: src/libraries/System.Data.Common/src/System.Data.Common.csproj [hunk 8]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <AssemblyName>System.Data.Common</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <NoWarn>$(NoWarn);CS1573</NoWarn>
+    <NoWarn>$(NoWarn);CS1573;CA2249</NoWarn>
```
Comment (stephentoub) [inline src/libraries/System.Data.Common/src/System.Data.Common.csproj:5]: Why was this rule disabled here? (Questioning it because this project only targets netcoreapp)

File: src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 9]
```diff
@@ -47,7 +47,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <NoWarn>$(NoWarn),0419,0649</NoWarn>
+    <NoWarn>$(NoWarn),0419,0649,CA2249,CA1830</NoWarn>
```
Comment (stephentoub) [inline src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj:50]: Why are these disabled for corelib?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs [hunk 10]
```diff
@@ -5489,26 +5489,26 @@ public void StartEvent(string eventName, EventAttribute eventAttribute)
             byteArrArgIndices = null;
 
             events.Append("  <event").
-                 Append(" value=\"").Append(eventAttribute.EventId).Append("\"").
-                 Append(" version=\"").Append(eventAttribute.Version).Append("\"").
-                 Append(" level=\"").Append(GetLevelName(eventAttribute.Level)).Append("\"").
-                 Append(" symbol=\"").Append(eventName).Append("\"");
+                 Append(" value=\"").Append(eventAttribute.EventId).Append('"').
```
Comment (danmosemsft) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:5492]: this isn't caught by https://github.com/dotnet/roslyn-analyzers/pull/3610 you just noticed it?

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs (hunk 10 above)
Comment (danmosemsft) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:5492]: Oh, I see your comment.

File: src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj (hunk 9 above)
Comment (pgovind) [reply src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj:50]: CA2249 is "Prefer string.Contains over string.IndexOf" and CA1830 is "Prefer typed StringBuilder overloads". Both these APIs are defined in corelib right? So they'd trigger the analyzer but there's nothing to do other than suppress them. I could make the suppression local if you'd prefer that

File: src/libraries/System.Data.Common/src/System.Data.Common.csproj (hunk 8 above)
Comment (pgovind) [reply src/libraries/System.Data.Common/src/System.Data.Common.csproj:5]: I honestly had the same questions. The CI build failed here without suppressing CA2249 even though the target says netcoreapp. I just assumed out build infra was also building this for another config somehow. Actually let's test it again now

File: src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj (hunk 7 above)
Comment (pgovind) [reply src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj:5]: Testing with the rule enabled
